src/topology/algebra/filter_basis.lean

```diff
@@ -348,6 +348,31 @@ def topology' {R M : Type*} [comm_ring R] {tR : topological_space R}
   [add_comm_group M] [module R M] (B : module_filter_basis R M) : topological_space M :=
   B.to_add_group_filter_basis.topology
 
+lemma _root_.has_continuous_smul.of_basis_zero {ι : Type*} [topological_ring R]
+  [topological_space M] [topological_add_group M] {p : ι → Prop} {b : ι → set M}
+  (h : has_basis (𝓝 0) p b) (hsmul : ∀ {i}, p i → ∃ (V ∈ 𝓝 (0 : R)) j (hj : p j), V • (b j) ⊆ b i)
+  (hsmul_left : ∀ (x₀ : R) {i}, p i → ∃ j (hj : p j), (b j) ⊆ (λ x, x₀ • x) ⁻¹' (b i))
+  (hsmul_right : ∀ (m₀ : M) {i}, p i → ∀ᶠ x in 𝓝 (0 : R), x • m₀ ∈ (b i)) :
+  has_continuous_smul R M :=
```

Comment:
```suggestion
lemma _root_.has_continuous_smul.of_basis_zero {ι : Type*} [topological_ring R]
  [topological_space M] [topological_add_group M] {p : ι → Prop} {b : ι → set M}
  (h : has_basis (𝓝 0) p b) (hsmul : ∀ {i}, p i → ∃ (V ∈ 𝓝 (0 : R)) j (hj : p j), V • (b j) ⊆ b i)
  (hsmul_left : ∀ (x₀ : R) {i}, p i → ∃ j (hj : p j), (b j) ⊆ (λ x, x₀ • x) ⁻¹' (b i))
  (hsmul_right : ∀ (m₀ : M) {i}, p i → ∀ᶠ x in 𝓝 (0 : R), x • m₀ ∈ (b i)) :
  has_continuous_smul R M :=
```
What's the difference between this and `module_filter_basis`?

Reply:
The `module_filter_basis` API is designed for defining a topology which is automatically a topological module, just from a filter basis satisfying some axioms. This versions says that if we have an existing add goup topology for which we know a basis of `𝓝 0`, and which satisfies some axioms, then we can get that this pre-existing topology is a module topology.

As I said this is mostly mathematically empty, but it's just easier to split the proof than to find a workaround for using the existing API.